front/lib/api/assistant/conversation.ts

```diff
@@ -859,7 +866,11 @@ export async function* postUserMessage(
     return;
   }
   // Check plan limit
-  const isAboveMessageLimit = await isMessagesLimitReached({ owner, plan });
+  const isAboveMessageLimit = await isMessagesLimitReached({
+    owner,
+    plan,
+    subscription,
+  });
```

Comment:
Can we adapt the error type and message below based on whether we are in trial or not :pray: